Fixes

GoChickenGo opened the pull request:
Shall we merge to master branch Tibbe?

thopp-tudelft committed:
fix line endings

thopp-tudelft committed:
remove unused imports and add TODOs

thopp-tudelft committed:
remove python 2 code

thopp-tudelft committed:
use importlib.resources instead of finding files

thopp-tudelft committed:
deSepcification

thopp-tudelft committed:
remove unused files

thopp-tudelft committed:
replace Icon with Logo

thopp-tudelft committed:
rename coordinatewidget

thopp-tudelft committed:
remove unused code

thopp-tudelft committed:
remove commented out code

thopp-tudelft committed:
remove comments in imports

thopp-tudelft committed:
remove more commented out code

thopp-tudelft committed:
autoformat

thopp-tudelft committed:
use isort --profile black -l 79 .

thopp-tudelft committed:
fix flake8 E711,E712 equality comparison on singleton

script:
mapfile -t files < <(find . -path "./[A-Za-z]*.py")
sed -i "${files[@]}" -e 's/and (\([A-Za-z0-9_.]\+\)):/and \1:/'
patterns=()
for keyword in if or and while; do
  for singleton in None True False; do
    patterns+=(-e 's/'"$keyword"' \([A-Za-z0-9_.]\+\) == \('"$singleton"'\)/'"$keyword"' \1 is \2/')
  done
done
sed -i "${files[@]}" "${patterns[@]}"
./format.sh

thopp-tudelft committed:
standardise comments, fix flake8 E262

script:
mapfile -t files < <(find . -path "./[A-Za-z]*.py")
sed -i "${files[@]}" -e 's/# *\t\+ */# /' -e 's/  #\+ */  # /' -e '2,$s/^#\+ */# /'
sed -i "${files[@]}" -e 's/ *\*\///'
sed -i "${files[@]}" -e 's/^\( *#\? \)[-=~#*]\{4,\} *\([^-=~#*]*[^-=~#* ]\) *[-=~#*]\{4,\}$/\1=== \2 ===/'
sed -i "${files[@]}" -e '/^ *#\? *[-=~#*]\{4,\}$/d'
./format.sh

thopp-tudelft committed:
fix flake8 E722 use of bare except:

script:
mapfile -t files < <(rg "^ *except:" --files-with-matches)
sed -i "${files[@]}" -e 's/^import .*/\0\nimport logging/' -e '/^\( *\)except:/{s//\1except Exception as exc:\n\1    logging.critical("caught exception", exc_info=exc)/;n;/^ *pass/d}'
./format.sh

thopp-tudelft committed:
remove assert True

thopp-tudelft committed:
setup basic logging to file

thopp-tudelft committed:
use fstrings instead of concatenating

thopp-tudelft committed:
replace print with logging.info

script:
mapfile -t files < <(rg "^ *print[(]" --files-with-matches)
sed -i "${files[@]}" -e 's/^import .*/\0\nimport logging/' \
  -e 's/\(^ *\)print(/\1logging.info(/'
./format.sh

thopp-tudelft committed:
fix remaining flake8 warnings

thopp-tudelft committed:
windows compatibility of the rename function

thopp-tudelft committed:
add excepthooks to log uncaught exceptions

thopp-tudelft committed:
unwavefrom

thopp-tudelft committed:
implement backwards compatible waveform_specification

thopp-tudelft committed:
update readme

thopp-tudelft committed:
move to installable package

moves all code to subdirectory
adds licence
adds pyproject.toml

thopp-tudelft committed:
add manual

thopp-tudelft committed:
fix reuse of mask function in dmdwidget

and refactors
fixes #6

thopp-tudelft commented:
I'd like to work on some of the regressions before merging: https://gitlab.tudelft.nl/gevidaq/gevidaq/-/issues

This doesn't necessarily block merging though, but we should keep a copy of the current master branch around while these issues are still up.

thopp-tudelft committed:
fix seal test output graph

in a previous refactor I noticed the QPen.setWidth(0.7) line and
corrected it to QPen.setWidthF to account for the floating point
value. the code actually relied on this value being truncated to 0.

this is now correctly documented in the code, next to other refactors
that allow one to actually read the code.

fixes #4

thopp-tudelft committed:
change aotf slider colors

thopp-tudelft committed:
fix wrong use of import inside function

fixes #3

thopp-tudelft committed:
additional formatting and refactors

thopp-tudelft committed:
set the aotf sliders to 0 when using the waveform function

fixes #7

thopp-tudelft committed:
use spinboxes in aotf widget

fixes #9

thopp-tudelft committed:
restore drop shadows in aotf widget

makes widget buttons look similar to old look again
see #9

thopp-tudelft commented:
:+1:
